[BugFix] follower sync dictionary refresh state

srlch opened the pull request:
## Why I'm doing:
In the current implementations, follower will not be synced any information about the dictionary refreshing which is a problem.

## What I'm doing:
Reuse dictionaryMgrInfo and DICTIONARY_MODIFY log to save the dictionary state information.

Fixes #issue
https://github.com/StarRocks/StarRocksTest/issues/7961

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

wyb commented on fe/fe-core/src/main/java/com/starrocks/catalog/Dictionary.java:
```diff
@@ -396,8 +423,6 @@ public List<String> getInfo() {
 
     @Override
     public void gsonPostProcess() throws IOException {
-        this.resetState();
-        this.updateNextSchedulableTime(this.getRefreshInterval());
     }
```
remove this method and `implements GsonPostProcessable`

srlch replied:
Fixed

wyb commented on fe/fe-core/src/main/java/com/starrocks/catalog/DictionaryMgr.java:
```diff
@@ -112,28 +109,47 @@ public class DictionaryMgr implements Writable, GsonPostProcessable {
 
-    // single thread execution
+    public void syncDictionaryMeta(List<Dictionary> dictionaries) {
+        if (dictionaries.size() == 0 ||
+                GlobalStateMgr.getCurrentState().getFeType() != FrontendNodeType.LEADER) {
```
```suggestion
        if (dictionaries.size() == 0 || !GlobalStateMgr.getCurrentState().isLeader()) {
```

srlch replied:
Fixed

wyb commented on fe/fe-core/src/main/java/com/starrocks/catalog/DictionaryMgr.java:
```diff
@@ -456,14 +480,30 @@ public void replayDropDictionary(String dictionaryName) {
+
+        // only replay for the follower sync
+        if (!GlobalStateMgr.isCheckpointThread() &&
+                GlobalStateMgr.getCurrentState().getFeType() == FrontendNodeType.FOLLOWER &&
```
or OBSERVER

srlch replied:
Fixed

srlch committed:
[BugFix] follower sync dictionary refresh state

Why I'm doing:
In the current implementations, follower will not be synced any
information about the dictionary refreshing which is a problem.

What I'm doing:
Reuse dictionaryMgrInfo and DICTIONARY_MODIFY log to save the
dictionary state information.

Signed-off-by: srlch <linzichao@starrocks.com>

srlch committed:
Fix fe ut

Signed-off-by: srlch <linzichao@starrocks.com>

srlch committed:
add common

Signed-off-by: srlch <linzichao@starrocks.com>
